### PR TITLE
Add send message button

### PR DESCRIPTION
**Pull request by SimonBrandner**
This PR adds a send button. Based on [this design](https://www.figma.com/file/DFBvBKfg2sZw6YrbqOYWK6/Composer?node-id=0%3A1). It fixes [#4963](https://github.com/vector-im/element-web/issues/4963) and fixes [#15611](https://github.com/vector-im/element-web/issues/15611).

Peek 2021-02-12 17-06
Peek 2021-02-12 17-23

### CODE REVIEW

**Comment by nadonomy** [comment]
Hey @SimonBrandner thanks for this! 

We've actually recently designed a similar change, which included some custom iconography and the icon/button changing state depending on input on the composer. Would you be ok to implement those changes? If so we can prepare assets for you next week.

**Comment by SimonBrandner** [comment]
Hi @nadonomy, sure!

**Comment by SimonBrandner** [comment]
@nadonomy I am wondering what's the view of the design team on [this](https://github.com/vector-im/element-web/issues/16416) (closing the emoji picker after an emoji was chosen). As this is another composer change I could include it in the PR with the other changes if you wanted.

**Comment by gaelledel** [comment]
Hi @SimonBrandner,

Thank you for your inputs!
I have some specs regarding the integration of the send button in the composer. Please have a look at this Figma file for more details: https://www.figma.com/file/DFBvBKfg2sZw6YrbqOYWK6/Composer?node-id=0%3A1
We'll have a default state in which the Send button isn't visible. The second state gets activated as soon as users are typing in the composer revealing the Send button. Give me a shout if you need anything.

Cheers,
Gaelle

**Comment by SimonBrandner** [comment]
Hi @gaelledel, thank you, I'll start working on this!

**Comment by SimonBrandner** [comment]
@gaelledel, one question. Should I remove the setting that allows you to always hide the send button?

**Comment by gaelledel** [comment]
@SimonBrandner, Yes please if you could remove this setting that would be best. Thanks a lot!

**Comment by SimonBrandner** [comment]
@gaelledel, how should the button exactly behave on hover? I don't think we have any similar icon button with a background in the app to compare this to. This is how it looks out of the box:
Screenshot_20210212_160826

By the way, the Figma design's sizes don't match up what we have currently:
Screenshot_20210212_161306
Screenshot_20210212_161235

**Comment by gaelledel** [comment]
> @gaelledel, how should the button exactly behave on hover? I don't think we have any similar icon button with a background in the app to compare this to. This is what it all looks like now (I diverged from the design a bit on the size of the button):
> Peek 2021-02-12 16-24
> 
> If you wanted to discuss this in a more real-time fashion you can DM me (`@simon.brandner:envs.net`)
> 
> By the way, the Figma design's sizes don't match up what we have currently:
> Screenshot_20210212_161306
> Screenshot_20210212_161235




@SimonBrandner No Problem, the button should be at 32px. We can dismiss any visual change on Hover just now.

**Comment by SimonBrandner** [comment]
@gaelledel, thank you for all your help with this! PR is now ready for review

File: src/components/views/rooms/SendMessageComposer.js [hunk 1]
```diff
@@ -536,10 +537,17 @@ export default class SendMessageComposer extends React.Component {
         }
     }
 
+    onChange = () => {
+        if (this.props.onIsEmptyChanged) {
```
Comment (jryans) [inline src/components/views/rooms/SendMessageComposer.js:541]: Hmm, at the moment, I think this is actually being called on _every_ change to the composer text, not only when the empty state changes. I guess this component could maintain some internal state so it only calls `onIsEmptyChanged` for an actual empty change, or alternatively you could call the props function just `onChange` so it's clear it receives the whole model for every change.

**Review by jryans — CHANGES_REQUESTED** [review]
Looks generally good code wise, just a small change to make.

**Review by gaelledel — APPROVED** [review]
All good provided the Send button has 16px padding on each side

**Review by turt2live — APPROVED** [review]
Looks great, thank you! Yet another PR of yours that solves something for an upcoming feature you weren't aware of 😇

**Comment by turt2live** [comment]
(I'll take care of merge conflicts, considering I created them)

**Pull request merged**